cmd/server-main.go

```diff
@@ -392,6 +392,8 @@ func serverMain(c *cli.Context) {
 	endpoints, err := parseStorageEndpoints(c.Args())
 	fatalIf(err, "Unable to parse storage endpoints %s", c.Args())
 
+	// Set endpoints of []*url.URL type to globalEndpoints.
+	globalEndpoints = endpoints
```

Comment:
Set this after all validation i.e move this below / after endpoints are considered to be sane and correct perhaps right before `initStorageDisks()`